### PR TITLE
[IMP] pivot: select dimension with arrow keys

### PR DESCRIPTION
**Pull request by LucasLefevre**
## Description:

description of this task, what is implemented and why it is implemented that way.

Task: : [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/composer/autocomplete_dropdown/autocomplete_dropdown_store.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  hide() {
+    this.provider = undefined;
+    this.selectedIndex = undefined;
+  }
+
+  selectIndex(index: number) {
+    this.selectedIndex = index;
+  }
+
+  selectUpOrDown(dir: "ArrowUp" | "ArrowDown") {
```
Comment (rrahir) [inline src/components/composer/autocomplete_dropdown/autocomplete_dropdown_store.ts:29]:
```suggestion
  moveSelection(direction: "previous" | "next") {
```
maybe?

**Review by rrahir — CHANGES_REQUESTED** [review]

File: src/components/composer/autocomplete_dropdown/autocomplete_dropdown_store.ts (hunk 1 above)
Comment (LucasLefevre) [reply src/components/composer/autocomplete_dropdown/autocomplete_dropdown_store.ts]: much better indeed

File: src/components/side_panel/pivot/pivot_dimensions/add_dimension_button/add_dimension_button.ts [hunk 2]
```diff
@@ -65,20 +104,41 @@ export class AddDimensionButton extends Component<Props, SpreadsheetChildEnv> {
...
+        const proposals = this.autoComplete.provider?.proposals;
+        if (proposals?.length === 1) {
+          this.autoComplete.provider?.selectProposal(proposals[0].text || "");
+        }
+        const proposal = this.autoComplete.selectedProposal;
+        this.autoComplete.provider?.selectProposal(proposal?.text || "");
+        break;
+      case "ArrowUp":
+      case "ArrowDown":
+        this.autoComplete.moveSelection(ev.key);
```
Comment (rrahir) [inline src/components/side_panel/pivot/pivot_dimensions/add_dimension_button/add_dimension_button.ts:135]:
```suggestion
        this.autoComplete.moveSelection(ev.key === "ArrowDown" ? "next" : "previous");
```

**Review by rrahir — CHANGES_REQUESTED** [review]
Could you add a test for the addDimensionButton?

File: src/components/side_panel/pivot/pivot_dimensions/add_dimension_button/add_dimension_button.ts (hunk 2 above)
Comment (LucasLefevre) [reply src/components/side_panel/pivot/pivot_dimensions/add_dimension_button/add_dimension_button.ts]: Oups, fixed :sweat_smile: 
The test is on odoo's side until pivots lands in o-spreadsheet
https://github.com/odoo/enterprise/pull/61671/files#diff-4b8b6f5d4dffedcb22e750296d4e58fb62e198e42789e82d1f1d6a28d4bdbc9dR511

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**